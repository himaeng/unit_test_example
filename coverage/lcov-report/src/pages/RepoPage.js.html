<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/pages/RepoPage.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/pages</a> RepoPage.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">91.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">91.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/12</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component,  } from 'react';
import { connect } from 'react-redux';
import { View, Text } from 'react-native';
&nbsp;
import RepoList from '../components/RepoList';
import { fetchData, selectRepo } from '../actions';
&nbsp;
const host = 'https://api.github.com';
const repositories =
  `${host}/search/repositories?q=react-native&amp;sort=stars&amp;order=desc`;
&nbsp;
export class RepoPage extends Component {
  static propTypes = {
    fetchData: React.PropTypes.func.isRequired,
    repos: React.PropTypes.object.isRequired,
    selectRepo: React.PropTypes.func,
    selected: React.PropTypes.bool
  }
&nbsp;
  componentDidMount() {
    this.props.fetchData(repositories);
  }
&nbsp;
  renderMainPart() {
    const { isLoading, repos, selected } = this.props.repos;
    if(!repos || isLoading) {
      return (
        &lt;Text style={styles.Loading}&gt;
          Loading...
        &lt;/Text&gt;
      );
    }
&nbsp;
    return (
      &lt;RepoList
        repos={repos}
        selected={selected}
        selectRepo={this.props.selectRepo}
      /&gt;
    );
  }
&nbsp;
  render() {
    const { isLoading, repos } = this.props.repos;
    return (
      &lt;View style={styles.container}&gt;
        {this.renderMainPart()}
      &lt;/View&gt;
    );
  }
}
&nbsp;
const styles = {
  container: {
    flex: 2,
    justifyContent: 'center'
  },
  loading: {
    fontSize: 20,
    alignSelf: 'center'
  }
}
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; {
<span class="cstat-no" title="statement not covered" >  return {</span>
    repos: state.repos
  }
}
&nbsp;
export default connect(
  mapStateToProps,
  { fetchData, selectRepo }
)(RepoPage);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Sat Apr 01 2017 15:58:50 GMT+0700 (ICT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
