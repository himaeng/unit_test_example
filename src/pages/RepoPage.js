import React, { Component } from 'react';
import { connect } from 'react-redux';
import { View, Text, TextInput, ScrollView } from 'react-native';

import RepoList from '../components/RepoList';
import { loadReposRequest, selectRepo } from '../actions';
import withKeyboard from '../enhancers/withKeyboard';

const host = 'https://api.github.com';
const repositories =
  `${host}/search/repositories?q=react-native&sort=stars&order=desc`;

export class RepoPage extends Component {
  static propTypes = {
    loadReposRequest: React.PropTypes.func.isRequired,
    repos: React.PropTypes.object.isRequired,
    selectRepo: React.PropTypes.func,
    selected: React.PropTypes.bool
  }

  componentDidMount() {
    this.props.loadReposRequest(repositories);
  }

  renderMainPart() {
    const { isLoading, repos, selected } = this.props.repos;
    if(!repos || isLoading) {
      return (
        <Text style={styles.Loading}>
          Loading...
        </Text>
      );
    }

    return (
      <ScrollView
        keyboardShouldPersistTaps='always'
        >
        <TextInput
          style={styles.search}
          placeholder='search'
          keyboardShouldPersistTaps='always'
        />
        <RepoList
          repos={repos}
          selected={selected}
          selectRepo={this.props.selectRepo}
          keyboardShouldPersistTaps='always'
        />
        <View style={{ height: this.props.keyboardHeight }} keyboardShouldPersistTaps='always' />
      </ScrollView>
    );
  }

  render() {
    const { isLoading, repos } = this.props.repos;
    return (
      <View style={styles.container} keyboardShouldPersistTaps='always'>
        {this.renderMainPart()}
      </View>
    );
  }
}

const styles = {
  container: {
    flex: 2,
    justifyContent: 'center'
  },
  loading: {
    fontSize: 20,
    alignSelf: 'center'
  },
  search: {
    backgroundColor: '#ededed',
    height: 25,
    marginTop: 70,
    marginHorizontal: 10
  }
}

const mapStateToProps = (state) => {
  return {
    repos: state.repos
  }
}

const RepoPageWithKeyboard = withKeyboard(RepoPage);

export default connect(
  mapStateToProps,
  { loadReposRequest, selectRepo }
)(RepoPageWithKeyboard);
