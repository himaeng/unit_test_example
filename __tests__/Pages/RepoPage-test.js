jest.mock('../../src/components/RepoList', () => 'Repolist');

import React from 'react';
import renderer from 'react-test-renderer';

import { RepoPage } from '../../src/pages/RepoPage';
import { repos } from '../../config/jest/mockData';

describe('RepoPage', () => {
  let getComponent = {};

  beforeEach(() => {
    getComponent = (props) => {
      return (
        renderer.create(
          <RepoPage
            repos={props}
            loadReposRequest={jest.fn()}
          />
        )
      )
    }
  });

  it('RepoPage จะ render เฉพาะ Loading text เมื่อ isLoading เป็น true และยังดึงข้อมูลไม่เสร็จ', () => {
    let component = getComponent({ isLoading: true, repos: [] });

    expect(component).toMatchSnapshot();
  });

  it('RepoPage จะ render เฉพาะ Loading text เมื่อ isLoading เป็น false และข้อมูล่โหลดเสร็จ', () => {
    let component = getComponent({ isLoading: false, repos });

    expect(component).toMatchSnapshot();
  });
});
