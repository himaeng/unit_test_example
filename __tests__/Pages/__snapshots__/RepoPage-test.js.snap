// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoPage RepoPage จะ render เฉพาะ Loading text เมื่อ isLoading เป็น false และข้อมูล่โหลดเสร็จ 1`] = `
<View
  style={
    Object {
      "flex": 2,
      "justifyContent": "center",
    }
  }
>
  <Repolist
    repos={
      Array [
        Object {
          "description": "Useful description goes here.",
          "id": 1,
          "name": "Test 1",
          "stargazers_count": 50000,
        },
        Object {
          "description": "Useful description goes here.",
          "id": 2,
          "name": "Test 2",
          "stargazers_count": 10000,
        },
        Object {
          "description": "Useful description goes here.",
          "id": 3,
          "name": "Test 3",
          "stargazers_count": 499,
        },
      ]
    }
  />
</View>
`;

exports[`RepoPage RepoPage จะ render เฉพาะ Loading text เมื่อ isLoading เป็น true และยังดึงข้อมูลไม่เสร็จ 1`] = `
<View
  style={
    Object {
      "flex": 2,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}
  >
    Loading...
  </Text>
</View>
`;
